### PR TITLE
[MM-37859] Fix product switcher menu items

### PR DESCRIPTION
**Pull request by deanwhillier**
#### Summary
The 'Download' and 'About' menu items were inadvertently added to the `Permissions.SYSCONSOLE_WRITE_PLUGINS` permissions gate when added during the Global Header push. This PR moves them back out so that they are accessible by all users and not just select admins.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37859

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by esethna — APPROVED** [review]

**Comment by matthewbirtch** [comment]
@deanwhillier I'm getting a blank screen on the test server again, so I can't review

File: components/global/product_switcher_menu/product_switcher_menu.tsx [hunk 1]
```diff
@@ -112,31 +112,33 @@ class ProductSwitcherMenu extends React.PureComponent<Props> {
...
+                        show={this.props.appDownloadLink && !UserAgent.isMobileApp()}
+                        url={useSafeUrl(this.props.appDownloadLink)}
+                        text={formatMessage({id: 'navbar_dropdown.nativeApps', defaultMessage: 'Download Apps'})}
+                        icon={
+                            <Icon
+                                size={16}
+                                glyph={'download-outline'}
+                            />
+                        }
+                        onClick={handleClick}
```
Comment (jespino) [inline components/global/product_switcher_menu/product_switcher_menu.tsx:127]: This looks like a problem with a bad merge.

File: components/widgets/menu/menu_items/menu_item_external_link.tsx [hunk 2]
```diff
@@ -17,7 +17,9 @@ export const MenuItemExternalLinkImpl: React.FC<Props> = ({url, text, onClick}:
         href={url}
         onClick={onClick}
     >
-        {text}
+        <span className='MenuItem__primary-text'>
```
Comment (jespino) [inline components/widgets/menu/menu_items/menu_item_external_link.tsx:20]: I don't know if this is right, but this is going to affect to any `ExternaLink` used, that can appear in other contexts.

File: components/global/product_switcher_menu/product_switcher_menu.tsx [hunk 3]
```diff
@@ -112,31 +112,33 @@ class ProductSwitcherMenu extends React.PureComponent<Props> {
...
+                        modalId={ModalIdentifiers.ABOUT}
+                        dialogType={AboutBuildModal}
+                        text={formatMessage({id: 'navbar_dropdown.about', defaultMessage: 'About {appTitle}'}, {appTitle: this.props.siteName})}
+                        icon={
+                            <Icon
+                                size={16}
+                                glyph={'information-outline'}
+                            />
+                        }
+                        onClick={handleClick}
```
Comment (jespino) [inline components/global/product_switcher_menu/product_switcher_menu.tsx:140]: Again, looks related to a bad merge.

**Review by jespino — CHANGES_REQUESTED** [review]
A couple of errors that looks related to a bad merge, and something that I'm not 100% sure if it is universally applicable.

File: components/widgets/menu/menu_items/menu_item_external_link.tsx (hunk 2 above)
Comment (deanwhillier) [reply components/widgets/menu/menu_items/menu_item_external_link.tsx:20]: Thanks for flagging! This fixes a visual inconsistency between external links and all other links.

I did double check all uses of the external menu item throughout the webapp and in each case, the buttons function as expected.

File: components/global/product_switcher_menu/product_switcher_menu.tsx (hunk 3 above)
Comment (deanwhillier) [reply components/global/product_switcher_menu/product_switcher_menu.tsx]: Thanks for catching!

File: components/global/product_switcher_menu/product_switcher_menu.tsx (hunk 1 above)
Comment (deanwhillier) [reply components/global/product_switcher_menu/product_switcher_menu.tsx]: Thanks for catching!

**Review by michelengelen — APPROVED** [review]

**Comment by deanwhillier** [comment]
@matthewbirtch, the test server is working now :)

**Review by matthewbirtch — APPROVED** [review]
Looks good to me @deanwhillier!

**Review by jespino — APPROVED** [review]

**Pull request merged**